include/swift/AST/DiagnosticsFrontend.def

```diff
@@ -157,6 +157,9 @@ ERROR(error_primary_file_not_found,none,
 ERROR(error_cannot_have_input_files_with_file_list,none,
 "cannot have input files with file list", ())
 
+ERROR(error_output_filenames_dont_match_primary_filenames,none,
+"in primary files mode the number of output filenames (%0) must match the number of input filenames (%1)", (unsigned, unsigned))
```

Comment:
This condition actually isn't specific to primary files mode, or whatever we're going to call it. In the existing per-file mode, you must have exactly one output for exactly one primary file. In WMO threaded mode, you must have exactly N outputs for exactly N inputs (none of which are "primary"). In WMO non-threaded mode, you must have exactly one output for exactly N inputs (none of which are primary). In this new mode, just having M outputs for M primary files won't be sufficient, because we have supplementary outputs as well.

Reply:
The message seems to match what the code is doing; so I'm not sure what you want to fix it here. Here is the predicate:
```C++
Opts.Inputs.havePrimaryInputsFilenames() &&
primaryFilenameCount != commandLineOutputFilenameCount &&
commandLineOutputFilenameCount != 0
```
Would you like to suggest a change to the diagnostic and/or a change to the predicate?

Reply:
Thinking about it a little more, I suspect that because of supplementary outputs, it won't make sense to check this at all in the final formulation of this code, which makes the condition you're checking only useful for WMO mode. I guess that means it matters less, but it's weird that we'd check this nicely for non-WMO mode and not for WMO mode.

Reply:
OK, I hope we can leave it for now.